src/main/java/org/matsim/contrib/gtfs/GtfsConverter.java

```diff
@@ -1,162 +1,226 @@
+            if (service.calendar_dates != null) {
+                for (LocalDate exceptionDate : service.calendar_dates.keySet()) {
+                    if (exceptionDate.isBefore(startDate)) {
+                        startDate = exceptionDate;
+                    }
+                }
+            }
+        }
+
+        log.info("Earliest date mentioned in feed: " + startDate);
+
+        LocalDate endDate = LocalDate.MIN;
+        for (Service service : this.feed.services.values()) {
+            if (service.calendar != null && service.calendar.end_date.isAfter(endDate)) {
+                endDate = service.calendar.end_date;
+            }
+            if (service.calendar_dates != null) {
+                for (LocalDate exceptionDate : service.calendar_dates.keySet()) {
+                    if (exceptionDate.isAfter(endDate)) {
+                        endDate = exceptionDate;
+                    }
+                }
+            }
+
+        }
+        log.info("Latest date mentioned in feed: " + endDate);
+
+        // Get the used service Id for the chosen weekday and date
+        List<String> activeServiceIds = this.getActiveServiceIds(this.feed.services);
+        log.info(String.format("Active Services: %d %s", activeServiceIds.size(), activeServiceIds));
+
+        // Get the Trips which are active today
+        List<Trip> activeTrips = feed.trips.values().stream()
+                .filter(trip -> feed.services.get(trip.service_id).activeOn(this.date))
+                .filter(this.filterTrips)
+                .filter(this::filterAgencyAndType)
+                .collect(Collectors.toList());
+
+        log.info(String.format("Active Trips: %d %s", activeTrips.size(), activeTrips.stream().map(trip -> trip.trip_id).collect(Collectors.toList())));
+
+        // Create one TransitLine for each GTFS-Route which has an active trip
+        activeTrips.stream().map(trip -> feed.routes.get(trip.route_id)).distinct().forEach(route -> {
+            TransitLine tl = ts.getFactory().createTransitLine(getReadableTransitLineId(route));
+            ts.addTransitLine(tl);
+            if (route.agency_id != null) tl.getAttributes().putAttribute("gtfs_agency_id", String.valueOf(route.agency_id));
+            tl.getAttributes().putAttribute("gtfs_route_type", String.valueOf(route.route_type)); // route type is a required field according to GTFS specification
+            String routeShortName = null;
+            if (route.route_short_name != null) {
+                routeShortName = route.route_short_name;
+            } else {
+                // use id in case there is no route short name
+                routeShortName = String.valueOf(route.route_id);
+            }
+            tl.getAttributes().putAttribute("gtfs_route_short_name",
+                    Normalizer.normalize(routeShortName, Normalizer.Form.NFD).replaceAll("[^\\p{ASCII}]", "")); // replaces non ascii symbols
+        });
+
+        this.convertTrips(activeTrips);
```

Comment:
I know you're operating on the scenario's transit schedule but could it be possible to chain this into the previous stream?

to me it looks like the foreach block is just another map and then the 'convertTrips' method would be in the foreach block. Adding the TransitLine to the schedule would be done later in this case. 

Just curious.